src/util.rs

```diff
@@ -228,3 +230,74 @@ pub fn short_file_path<'a>(target: &'a str, current_dir: &str) -> &'a str {
         .and_then(|p| p.to_str())
         .unwrap_or(target)
 }
+
+/// Given a starting Location, try to find `name` in a file and report its Position
+///
+/// If `treat_name_as_symbol` is true, then we're not doing a pure text search but
+/// treating name as a symbol. We apply a crude heuristic to ensure we have found
+/// the `name` symbol instead of `name` as part of a larger string.
+///
+/// If `treat_name_as_symbol` is false then this a pure search for a string in filename.
+pub fn find_name_in_file(
+    filename: &str,
+    ctx: &Context,
+    start_from: Position,
+    name: &str,
+    treat_name_as_symbol: bool,
+) -> Option<Position> {
+    if name.is_empty() {
+        return Some(start_from);
+    }
+    let contents = get_file_contents(filename, ctx).unwrap();
+    let line_offset: usize = contents.line_to_char(start_from.line.try_into().unwrap());
+    let within_line_offset: usize = start_from.character.try_into().unwrap();
+    let char_offset = line_offset + within_line_offset;
+    let mut it = name.chars();
+    let mut maybe_found = None;
+    for (num, c) in contents.chars_at(char_offset).enumerate() {
```

Comment:
I don't understand everything here, but this looks like it could be done with a simple regex search.
`treat_name_as_symbol` is effectively unused right now (since it's always true), so we should remove it.
With that, we can use a `\b` word boundary instead of checking if the next character is alphanumeric,
that will protect against `f_totally_not_f`.

Reply:
I don't know how we can do a regex search on Ropey without rendering the whole document to a `String` (which I've tried to avoid...)

Reply:
Ok that makes sense (I hadn't checked in detail).
Actually, since we now ask for DocumentSymbol, do we still need the workarounds for SymbolInformation?
Is there any relevant server that still only returns the deprecated SymbolInformation?
If not, then I'd just go with directly returning what the server gives us, to make everything more predictable.

Reply:
I don't think we can assume that all LSPs will return `DocumentSymbol`. Let me check the LSPs for which I had added workarounds. If they are now returning `DocumentSymbols` I will remove workarounds for them.

Reply:
I removed the special case for elixir-ls as it is now sending DocumentSymbol. Erlang still sends SymbolInformation so special case for that remains.

However at the current moment `textDocument/documentSymbol` is somewhat broken on this branch and master. See https://github.com/kak-lsp/kak-lsp/pull/557#issuecomment-969154614